x-pack/solutions/observability/plugins/synthetics/server/services/monitor_config_repository.ts

```diff
@@ -185,18 +192,71 @@ export class MonitorConfigRepository {
     monitors: Array<{
       attributes: MonitorFields;
       id: string;
-      soType: string;
+      previousMonitor: SavedObject<SyntheticsMonitorWithSecretsAttributes>;
     }>;
     namespace?: string;
   }) {
-    return this.soClient.bulkUpdate<MonitorFields>(
-      monitors.map(({ attributes, id, soType }) => ({
-        type: soType,
+    // Split monitors into those needing recreation and those that can be updated
+    const toRecreate: Array<{
+      id: string;
+      attributes: MonitorFields;
+      previousMonitor: SavedObject<SyntheticsMonitorWithSecretsAttributes>;
+    }> = [];
+    const toUpdate: Array<{
+      type: string;
+      id: string;
+      attributes: MonitorFields;
+      namespace?: string;
+    }> = [];
+
+    for (const monitor of monitors) {
+      const { attributes, id, previousMonitor } = monitor;
+      const prevSpaces = (previousMonitor.namespaces || []).sort();
+      const spaces = (attributes.spaces || []).sort();
+      if (!isEqual(prevSpaces, spaces) && !isEmpty(spaces)) {
```

Comment:
I don't think we should recreate the monitor everytime that spaces changes. We should do it only if a single space monitor is converted to a multi space one.